agent/event_endpoint.go

```diff
@@ -127,25 +127,44 @@ RUN_QUERY:
 	// Get the recent events
 	events := s.agent.UserEvents()
 
+	// Filter the events if requested
+	if nameFilter != "" {
+		for i := 0; i < len(events); i++ {
+			if events[i].Name != nameFilter {
+				events = append(events[:i], events[i+1:]...)
+				i--
+			}
+		}
+	}
```

Comment:
Just curious, why are we doing the name filter before the ACL filter?

Reply:
See comment below (on L142) 😄 

```Go
// Note: we filter the results with ACLs *after* applying the user-supplied
// name filter, to ensure the filtered-by-acls header we set below does not
// include results that would be filtered out even if the user did have
// permission.
```